### PR TITLE
Homework View

### PR DESCRIPTION
**Pull request by Vito0912**
This is based on #328.

It adds a button to the app bar to switch between the homework view and the lesson view. The homework view displays all items with homework, sorted by undone and latest date.
So oldest undone homework will be displayed first.

grafik

This PR also fixes an issue with long course names overflowing.

### CODE REVIEW

File: app/lib/applets/lessons/student/lessons_student_view.dart [hunk 1]
```diff
@@ -15,65 +16,126 @@ class LessonsStudentView extends StatefulWidget {
...
+            });
+          } else {
+            attendanceLessons = lessons
+                .where((element) => element.attendances != null)
+                .toList();
+          }
 
           return Scaffold(
+            appBar: widget.openDrawerCb != null
+                ? AppBar(
```
Comment (alessioC42) [inline app/lib/applets/lessons/student/lessons_student_view.dart:73]: The app bar has to be located outside of the CombinedAppletBuilder, to ensure it is rendered and the user has the ability to open the sidebar while on a loading screen.

File: app/lib/applets/lessons/student/lessons_student_view.dart [hunk 2]
```diff
@@ -15,65 +16,126 @@ class LessonsStudentView extends StatefulWidget {
...
+                    title: Text(settings['showHomework'] == 'true'
+                        ? AppLocalizations.of(context)!.homework
```
Comment (alessioC42) [inline app/lib/applets/lessons/student/lessons_student_view.dart:75]: I would not change the AppBar title design wise. But rather show a snackbar with "Showing Homework" or "Showing all courses"

File: app/lib/applets/lessons/student/lessons_student_view.dart [hunk 3]
```diff
@@ -15,65 +16,126 @@ class LessonsStudentView extends StatefulWidget {
...
+                        : lessonsDefinition.label(context)),
+                    leading: IconButton(
```
Comment (alessioC42) [inline app/lib/applets/lessons/student/lessons_student_view.dart:77]: We should hide the ActionButton if there is no Homework

File: app/lib/applets/lessons/student/lessons_student_view.dart [hunk 4]
```diff
@@ -15,65 +16,126 @@ class LessonsStudentView extends StatefulWidget {
...
+                      icon: const Icon(Icons.menu),
+                      onPressed: () => widget.openDrawerCb!(),
+                    ),
+                    actions: [
```
Comment (alessioC42) [inline app/lib/applets/lessons/student/lessons_student_view.dart:81]: We should hide the actions when there is no homework

**Review by alessioC42 — CHANGES_REQUESTED** [review]
Overall a solid feature, but please fix the issues marked

File: app/lib/applets/lessons/student/lessons_student_view.dart (hunk 4 above)
Comment (Vito0912) [reply app/lib/applets/lessons/student/lessons_student_view.dart:81]: We can do this, but I want to add that homework is something that changes every week or day.

Therefore, I would vote to always display this button (maybe with a different message, like "Currently no homework"). However, if the user has no homework for a week, they would not have the possibility to switch back to the page once they have viewed all the courses. And either way the user will view an empty page if that was the last page, but he would not be able to switch back

Opinion?

Comment (alessioC42) [reply app/lib/applets/lessons/student/lessons_student_view.dart:81]: So check if the setting is causing the app to display nothing and turn it of?

**Review by alessioC42 — APPROVED** [review]

**Pull request merged**